File: MyAesCipher.php
<?php

/**
 * AesCipher
 *
 * Encode/Decode text by password using AES-128-CBC algorithm
 */
class AesCipher
{
    const CIPHER = 'AES-128-CBC';
    const INIT_VECTOR_LENGTH = 16;

    /**
     * AesCipher constructor.
     *
     */
    public function __construct()
    {

    }

    public static function encryptStrAndToBase64($keyStr, $enStr)
    {
      try {
        // $result = encrypt($keyStr, $keyStr, utf8_encode($enStr));
        $ivBytes = substr(md5($keyStr), 0, static::INIT_VECTOR_LENGTH);
        $keyBytes = hash('sha256', $keyStr);
        $enBytes = utf8_encode($enStr);        
        return static::encrypt($ivBytes, $keyBytes, $enBytes);
      } catch (\Exception $e) {
        return $e->getMessage();
      }
    }

    public static function encrypt($ivBytes, $keyBytes, $bytes) {
      try {
        // Encrypt input text
        $raw = openssl_encrypt(
            $bytes,
            static::CIPHER,
            $keyBytes,
            OPENSSL_RAW_DATA,
            $ivBytes
        );

        $result = base64_encode($raw);

        if ($result === false) {
            // Operation failed
            return new static($ivBytes, null, openssl_error_string());
        }

        // Return successful encoded object
        return $result;
      } catch (\Exception $e) {
          // Operation failed
          return new static(isset($ivBytes), null, $e->getMessage());
      }
    }

    public static function decryptStrAndFromBase64($keyStr, $deStr) 
    {
        try {
            $deBytes = base64_decode(utf8_decode($deStr));
            $ivBytes = substr(md5($keyStr), 0, static::INIT_VECTOR_LENGTH);
            $keyBytes = hash('sha256', $keyStr);
            return static::decrypt($ivBytes, $keyBytes, $deBytes);
        } catch (\Exception $e) {
            return $e->getMessage();
        }
    }

    public static function decrypt($ivBytes, $keyBytes, $deBytes)
    {
        try {
            $decoded = openssl_decrypt(
                $deBytes,
                static::CIPHER,
                $keyBytes,
                OPENSSL_RAW_DATA,
                $ivBytes
            );
    
            if ($decoded === false) {
                // Operation failed
                return new static(isset($ivBytes), null, openssl_error_string());
            }

            // Return successful decoded object
            return $decoded;
        } catch (\Exception $e) {
            // Operation failed
            return new static(isset($ivBytes), null, $e->getMessage());
        }
    }

}

// USAGE

$secretKey = '26kozQaKwRuNJ24t';
$text = 'Some text';

$encrypted = AesCipher::encryptStrAndToBase64($secretKey, $text);
$decrypted = AesCipher::decryptStrAndFromBase64($secretKey, $encrypted);

echo($encrypted);
echo("</br>");
echo($decrypted);

File: MyAesCipher.java
import java.util.Base64;
import java.security.MessageDigest;
import javax.crypto.Cipher;
import javax.crypto.spec.*;
import java.security.spec.AlgorithmParameterSpec;
import java.lang.Exception;

public class MyAesCipher{
	static void myMethod() {
		//		encryptStrAndToBase64("26kozQaKwRuNJ24t", "result");
		try {
			String encrypted = encryptStrAndToBase64("26kozQaKwRuNJ24t", "Some text");
			String decrypted = decryptStrAndFromBase64("26kozQaKwRuNJ24t", encrypted);
			System.out.println(encrypted);
			System.out.println(decrypted);
		} catch (Exception e){
			System.out.println("Exception Error: " + e);
		}
	}

	public static void main(String[] args) {
		myMethod();
	}

	public static String encryptStrAndToBase64(String keyStr, String enStr) throws Exception{
		byte[] bytes = encrypt(keyStr, keyStr, enStr.getBytes("UTF-8"));
		return new String(Base64.getEncoder().encode(bytes), "UTF-8");
	}  

	public static String decryptStrAndFromBase64(String keyStr, String deStr) throws Exception{
		byte[] bytes = decrypt(keyStr, keyStr, Base64.getDecoder().decode(deStr.getBytes("UTF-8")));
		return new String(bytes, "UTF-8");
	}

	public static byte[] encrypt(String ivStr, String keyStr, byte[] bytes) throws Exception{
		MessageDigest md = MessageDigest.getInstance("MD5");
		md.update(ivStr.getBytes());
		byte[] ivBytes = md.digest();

		MessageDigest sha = MessageDigest.getInstance("SHA-256");
		sha.update(keyStr.getBytes());
		byte[] keyBytes = sha.digest();

		return encrypt(ivBytes, keyBytes, bytes);
	}

	static byte[] encrypt(byte[] ivBytes, byte[] keyBytes, byte[] bytes) throws Exception{
		AlgorithmParameterSpec ivSpec = new IvParameterSpec(ivBytes);
		SecretKeySpec newKey = new SecretKeySpec(keyBytes, "AES");
		Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
		cipher.init(Cipher.ENCRYPT_MODE, newKey, ivSpec);
		return cipher.doFinal(bytes);
	}

	public static byte[] decrypt(String ivStr, String keyStr, byte[] bytes) throws Exception{
		MessageDigest md = MessageDigest.getInstance("MD5");
		md.update(ivStr.getBytes());
		byte[] ivBytes = md.digest();

		MessageDigest sha = MessageDigest.getInstance("SHA-256");
		sha.update(keyStr.getBytes());
		byte[] keyBytes = sha.digest();

		return decrypt(ivBytes, keyBytes, bytes);
	}

	static byte[] decrypt(byte[] ivBytes, byte[] keyBytes, byte[] bytes)  throws Exception{
		AlgorithmParameterSpec ivSpec = new IvParameterSpec(ivBytes);
		SecretKeySpec newKey = new SecretKeySpec(keyBytes, "AES");
		Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
		cipher.init(Cipher.DECRYPT_MODE, newKey, ivSpec);
		return cipher.doFinal(bytes);
	}
}
